src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -43,7 +43,14 @@ public sealed class NodeHandler : Handler, INodeHandler
             string workingDirectory = Data.WorkingDirectory;
             if (string.IsNullOrEmpty(workingDirectory))
             {
-                workingDirectory = TaskDirectory;
+                if (!string.IsNullOrEmpty(ExecutionContext.Variables.System_DefaultWorkingDirectory))
+                {
```

Comment:
When would this not be set?  Couldn't this actually be an assert since it should never not be set?